proposals/infra/INF-0002-root-signature-in-clang.md

```diff
@@ -304,15 +304,147 @@ elimation has completed.
 
 ### Validations in Sema
 
-TODO
+#### All the values should be legal.
+
+  ShaderVisibility
+
+      'SHADER_VISIBILITY_ALL'
+      'SHADER_VISIBILITY_VERTEX'
+      'SHADER_VISIBILITY_HULL'
+      'SHADER_VISIBILITY_DOMAIN'
+      'SHADER_VISIBILITY_GEOMETRY'
+      'SHADER_VISIBILITY_PIXEL'
+      'SHADER_VISIBILITY_AMPLIFICATION'
+      'SHADER_VISIBILITY_MESH'
+
+  ParameterType
+
+      CBV
+      SRV
+      UAV
+      DescriptorTable
+      Constants32Bit
+
+  RootDescriptorFlags
+
+      None
+      DataVolatile
+      DataStaticWihleSetAtExecute
+      DataStatic
+
+  DescriptorRangeFlags
+
+    For Sampler
+      None
+      DescriptorsVolatile
+      DESCRIPTORS_STATIC_KEEPING_BUFFER_BOUNDS_CHECKS
+
+    For resource
+      None
+      DESCRIPTORS_VOLATILE,
+      DATA_VOLATILE,
+      DATA_STATIC,
+      DATA_STATIC_WHILE_SET_AT_EXECUTE,
+      DESCRIPTORS_VOLATILE | DATA_VOLATILE,
+      DESCRIPTORS_VOLATILE | DATA_STATIC_WHILE_SET_AT_EXECUTE,
+
+      DESCRIPTORS_STATIC_KEEPING_BUFFER_BOUNDS_CHECKS,
+      DESCRIPTORS_STATIC_KEEPING_BUFFER_BOUNDS_CHECKS | DATA_VOLATILE,
+      DESCRIPTORS_STATIC_KEEPING_BUFFER_BOUNDS_CHECKS | DATA_STATIC,
+      DESCRIPTORS_STATIC_KEEPING_BUFFER_BOUNDS_CHECKS | DATA_STATIC_WHILE_SET_AT_EXECUTE,
+
+  StaticSampler
+
+      Max/MinLOD cannot be NaN.
+      Comparison filter must have ComparisonFunc not equal to 0.
+      MaxAnisotropy cannot exceed 16.
+      MipLODBias must be within range of [-16, 15.99].
+
+  Space
+
+      The range 0xFFFFFFF0 to 0xFFFFFFFF is reserved.
+      "CBV(b0, space=4294967295)" is invalid duo to the use of reserved space 0xFFFFFFFF.
+
+#### Resource ranges must not overlap.
+
+      "CBV(b2), DescriptorTable(CBV(b0, numDescriptors=5))" will result in an 
+      error due to overlapping at b2.
+
 
 ### Metadata Schema
 
-TODO
+Here is the metadata presentation for the root signature.
+
+Root Signature
+| | Version | Flag | RootParameters | StaticSamplers |
+|-| --- | --- | --- | --- |
+| | i32 | i32 | list of RootParameters (RootConstant, RootDescriptor, DescriptorTable) | list of StaticSamplers |
```

Comment:
This doesn't match the format that was proposed previously.  See the "Root Signatures in the LLVM IR" section above.

Also, note that there's a separate issue for updating the metadata schema.  I suggest you tackle the validation and metadata changes in different PRs.

Reply:
Removed.